File: backend/src/AnimeShop/API/Stock.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE TypeOperators #-}
module AnimeShop.API.Stock (StockAPI, stockServer) where

import AnimeShop.Prelude
import AnimeShop.API.Stock.Items
import AnimeShop.API.Stock.ItemInstances
import AnimeShop.API.Stock.Manufacturers

type StockAPI = "stock" :>
  (ItemsAPI :<|> ItemInstancesAPI :<|> ManufacturersAPI)

stockServer :: ServerT StockAPI App
stockServer = itemsServer :<|> itemInstancesServer :<|> manufacturersServer

File: frontend/src/components/Pages/Stock.js
import React from "react";
import Manufacturers from "../Stock/Manufacturers";
import Items from "../Stock/Items";
import ItemInstances from "../Stock/ItemInstances";

export function Stock() {
    return (
      <div>
        <Manufacturers/>
        <Items/>
        <ItemInstances/>
      </div>
    );
  }
